File: backend/Application/Mappers/TechnologyMapper.cs
using Application.DTOs.Responses;
using Application.Interfaces;
using Domain.Entities;

namespace Application.Mappers;

public class TechnologyMapper : ITechnologyMapper
{
    public TechnologyResponse Map(Technology technology)
    {
        return new TechnologyResponse
        {
            Id = technology.Id,
            Name = technology.Name,
            Usage = technology.Usage.ToString(),

            Media = technology.Media is null
                ? null
                : new MediaResponse
                {
                    Path = technology.Media.Path,
                    AltText = technology.Media.AltText,
                    Type = technology.Media.Type
                }
        };
    }
}

File: frontend/src/mappers/technologyMapper.js
export function mapTechnology(data) {
    if (!data) {
        throw new Error("Technology data is missing");
    }

    return {
        id: data.id,
        name: data.name ?? "",
        usage: data.usage ?? "",
        media: data.media
            ? {
                path: data.media.path ?? "",
                altText: data.media.altText ?? "",
                type: data.media.type,
            }
            : null,
    };
}